readthedocs/projects/tasks.py

```diff
@@ -1005,3 +1007,15 @@ def sync_callback(_, version_pk, commit, *args, **kwargs):
     """
     fileify(version_pk, commit=commit)
     update_search(version_pk, commit=commit)
+
+
+@app.task()
+def finish_inactive_builds():
+    query = (~Q(state=BUILD_STATE_FINISHED) &
+             Q(date__lte=datetime.datetime.now() - datetime.timedelta(minutes=45)))
+    # TODO: consider ``poject.container_time_limit`` since it could be bigger than 45 minutes
+    for build in Build.objects.filter(query):
```

Comment:
This could be `build.update()` instead.

Reply:
If we are going to use `update` we can't put the `build id #` in the message.

So, for now, I'm adding this message:

```
        error=('This build was terminated due to inactivity. If you continue '
               'to encounter this error, file a support request.')
```

Reply:
Also, I think I will limit the query to the first 25 or 50 elements because otherwise the first time this task will be ran it will take tons of existing builds from all the times...

Reply:
Besides, this is not valid...

    AssertionError: Cannot update a query once a slice has been taken.

So, maybe go back to my approach is better.